aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -486,6 +486,13 @@ std::unique_ptr<TensorIterator> TensorIterator::unary_op(Tensor& out, const Tens
   return builder.build();
 }
 
+std::unique_ptr<TensorIterator> TensorIterator::nullary_op(Tensor& out) {
```

Comment:
Umm... are you sure? Isn't this just an in-place unary op?

cc @colesbury

Reply:
It takes no Tensor arguments so it's nullary. `f() -> [random]`. As opposed to something like `x.pow_(2)` which is the inplace version of a unary function: `f(x) -> x^2`. (The naming is unfortunate and I'm probably playing fast and loose with the terminology, but... oh well)

Since it doesn't read `out` it shouldn't be added as an input below.

Reply:
I did try not adding `out` as an input, but got the weird behavior below where I keep getting one element tensor as the output, even though I asked for more:
```
Python 3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:22:34) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> torch.FloatTensor(10).cuda().uniform_()
tensor(0.7869, device='cuda:0')
>>> torch.FloatTensor(1000).cuda().uniform_()
tensor(0.7470, device='cuda:0')
>>> torch.FloatTensor(10000).cuda().uniform_()
tensor(0.0427, device='cuda:0')
>>> 
```
Adding out as an input fixed it. @colesbury do you know why?

Reply:
Oh, that sounds like a bug in TensorIterator. The problem is the shape computation, which tries to preserve some legacy resizing behavior and it doesn't work when there's only one Tensor.

The easiest workaround is copy this line to prevent resizing the output. I think that's better than adding the Tensor as an input.

https://github.com/pytorch/pytorch/blob/558c6c4d8ab76f1c8e075a230af584258c9c89e8/aten/src/ATen/native/TensorIterator.cpp#L495

We'll fix the problem in TensorIterator more generally later.